File: app/models/teacher.rb
class Teacher < ApplicationRecord
  belongs_to :user
  has_many :courses
  # has_and_belongs_to_many :students

  has_many :teacher_students
  has_many :students, :through => :teacher_students

  mount_uploaders :audio, AudioUploader
  mount_uploaders :resume, FileUploader
  mount_uploaders :certificates, CertificateUploader

  def active?
    return self.status == "active"
  end

  def pending?
    self.status == "pending"
  end

  def fullname
    return self.lastname + self.firstname
  end
end

File: models/teacher.js
/* 
 * @author: znz
*/

var mongoose = require('mongoose');

var teacherSchema = new mongoose.Schema({
  title: String,
  description: String,
  author: String,
  firstname: String,
  lastname: String,
  englishname: String,
  age: Number,
  birthday: String,
  gender: String,
  city: String,
  work: String,
  education: String,
  experience: String,
  otherexperience: String,
  profour: Number,
  proeight: Number,
  levelsix: Number,
  other: String,
  honor: String,
  interaction: String,
  like: String,
  availabletime: Number,
  audio: String,
  comments: String,
  resume: String,
  level: String,
  status: { type: String, default: "pending" },
  certificates: [],
  students: [ {type: mongoose.Schema.Types.ObjectId, ref: 'Student'} ],
  courses: [ {type: mongoose.Schema.Types.ObjectId, ref: 'Course'} ],
  user_id: { type: mongoose.Schema.Types.ObjectId, ref: 'User' },
  created_at: Date,
  updated_at: Date
});

teacherSchema.pre("save", function(next){
  var currentDate = new Date();
  this.updated_at = currentDate;
  if ( !this.created_at ) {
    this.created_at = currentDate;
  }
  next();
});

var Teacher = mongoose.model('Teacher', teacherSchema);

module.exports = Teacher;